api/public/public_api/endpoints/audio_transcription.py

```diff
@@ -70,93 +64,71 @@ def _transcribe(self) -> Optional[str]:
-        request_headers = {
-            "authorization": environ["STT_API_KEY"],
-            "content-type": "application/json",
-        }
+        client = speech.SpeechClient()
+        audio = speech.RecognitionAudio(content=self.request.data)
+        config_values = dict(
+            encoding=speech.RecognitionConfig.AudioEncoding.FLAC,
+            sample_rate_hertz=SAMPLE_RATE,
+            language_code="en-US",
+        )
+        config = speech.RecognitionConfig(**config_values)
         start_timestamp = datetime.now()
         try:
-            response = requests.post(
-                environ["STT_URL"],
-                headers=request_headers,
-                data=request_data,
-                timeout=5,
-            )
-            response.raise_for_status()
-        except requests.ConnectionError:
-            _log.exception(
-                f"{self.request_id}: Failed to connect to audio transcription service"
-            )
-        except requests.HTTPError:
-            log_message = (
-                f"{self.request_id}: API request to transcription service failed"
-            )
-            response_text = json.loads(response.text)
-            error_message = response_text.get("error")
-            if error_message is not None:
-                log_message += f": {error_message}"
-            _log.exception(log_message)
+            response = client.recognize(config=config, audio=audio)
+        except Exception:
+            _log.exception(f"{self.request_id}: Transcription failed.")
         finally:
             end_timestamp = datetime.now()
         transcription_duration = (end_timestamp - start_timestamp).total_seconds()
-        self.transcription_duration = Decimal(transcription_duration)
+        self.transcription_duration = Decimal(str(transcription_duration))
 
         return response
 
-    def _handle_api_response(
-        self, response: Optional[requests.Response]
+    def _get_transcription(
+        self, response: Optional[speech.RecognizeResponse]
     ) -> Optional[str]:
         """Interrogates the response from the transcription service API.
 
         :param response: the transcription service API response
         :return: None if the audio could not be transcribed or the transcription
         """
         transcription = None
-        if response is not None:
-            response_data = json.loads(response.text)
-            if response_data["status"] == "completed":
-                self.transcription_success = True
-                transcription = response_data["text"]
-            else:
-                _log.warning(f"{self.request_id}: audio could not be transcribed")
+        if response:
+            highest_confidence = 0
+            for result in response.results:
+                for alternative in result.alternatives:
+                    if alternative.confidence > highest_confidence:
+                        transcription = alternative.transcription
```

Comment:
How many of these alternatives does it usually return?
I'm assuming not many, just curious.

Reply:
I have only ever seen one